engine/paymail/paymail_service_client.go

```diff
@@ -0,0 +1,194 @@
+package paymail
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/bitcoin-sv/go-paymail"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"github.com/mrz1836/go-cachestore"
+	"github.com/rs/zerolog"
+)
+
+const cacheKeyCapabilities = "paymail-capabilities-"
+const cacheTTLCapabilities = 60 * time.Minute
+
+type service struct {
+	cache         cachestore.ClientInterface
+	paymailClient paymail.ClientInterface
+	log           zerolog.Logger
+}
+
+// NewServiceClient creates a new paymail service client
+func NewServiceClient(cache cachestore.ClientInterface, paymailClient paymail.ClientInterface, log zerolog.Logger) ServiceClient {
+	if paymailClient == nil {
+		panic(spverrors.Newf("paymail client is required to create a new paymail service"))
+	}
+
+	if cache == nil {
+		log.Info().Msg("Doesn't receive cachestore, won't use cache for capabilities")
+	}
```

Comment:
Is it possible in our project, that we have no cache service?
Maybe during tests it is nil - but... as I recall we always have cache service (redis or freecache)

Reply:
Good question,
I don't know if this is possible, It was done this way previously so I assume that it's possible. I will try to check that.